pkg/asset/installconfig/gcp/session.go

```diff
@@ -0,0 +1,173 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	googleoauth "golang.org/x/oauth2/google"
+	compute "google.golang.org/api/compute/v1"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+var (
+	authEnvs            = []string{"GOOGLE_CREDENTIALS", "GOOGLE_CLOUD_KEYFILE_JSON", "GCLOUD_KEYFILE_JSON"}
+	defaultAuthFilePath = filepath.Join(os.Getenv("HOME"), ".gcp", "osServiceAccount.json")
+)
+
+// Session is an object representing session for GCP API.
+type Session struct {
+	Credentials *googleoauth.Credentials
+}
+
+// GetSession returns an GCP session by using credentials found in default locations in order:
+// env GOOGLE_CREDENTIALS,
+// env GOOGLE_CLOUD_KEYFILE_JSON,
+// env GCLOUD_KEYFILE_JSON,
+// file ~/.gcp/osServiceAccount.json, and
+// gcloud cli defaults
+// and, if no creds are found, asks for them and stores them on disk in a config file
+func GetSession(ctx context.Context) (*Session, error) {
+	creds, err := loadCredentials(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load credentials")
+	}
+
+	return &Session{
+		Credentials: creds,
+	}, nil
+}
+
+func loadCredentials(ctx context.Context) (*googleoauth.Credentials, error) {
+	var loaders []credLoader
+	for _, env := range authEnvs {
+		loaders = append(loaders, &envLoader{env: env})
+	}
+	loaders = append(loaders, &fileLoader{path: defaultAuthFilePath})
+	loaders = append(loaders, &cliLoader{})
+
+	for _, l := range loaders {
+		creds, err := l.Load(ctx)
+		if err != nil {
+			logrus.Debug(errors.Wrapf(err, "failed to load credentials from %s", l))
+			continue
+		}
+		return creds, nil
+	}
+	return getCredentials(ctx)
+}
+
+func getCredentials(ctx context.Context) (*googleoauth.Credentials, error) {
+	creds, err := (&userLoader{}).Load(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	filePath := defaultAuthFilePath
+	logrus.Info("Saving the credentials to %q", filePath)
```

Comment:
For all platforms, the installer expects the user provide credentials from the environmentm but in cases where the installer fails to load any, the requirement has been to prompt the user for creds and store them in an location such that future calls to the installer succeed without any prompt. Therefore, when user was prompted for creds here, we save it to ~/.gcp/osServiceAccount.json so that future installer calls work without prompts.

INFO should be enough for users to know where there creds were stored.